api/ncf_api_flask_app/views.py

```diff
@@ -74,59 +83,89 @@ def get_path_from_args (request):
-  resp = jsonify( techniques )
-  return resp
+  try:
+    # We need to get path from url params, if not present put "" as default value
+    path = get_path_from_args(request)
+  
+    techniques = ncf.get_all_techniques_metadata(alt_path = path)
+    resp = jsonify( techniques )
+    return resp
+  except :
+    errstr = traceback.format_exc()
+    sys.stderr.write(errstr)
+    sys.stderr.flush()
+    error = jsonify ({ "error": "Unknown internal error", "details": errstr})
+    error.status_code = 500
+    return  error
 
 @app.route('/api/generic_methods', methods = ['GET'])
 @check_auth
 def get_generic_methods():
   """Get all generic methods from ncf folder passed as parameter, or default ncf folder if not defined"""
-  # We need to get path from url params, if not present put "" as default value
-  path = get_path_from_args(request)
-
-  generic_methods = ncf.get_all_generic_methods_metadata(alt_path = path)
-  resp = jsonify( generic_methods )
-  return resp
+  try:
+    # We need to get path from url params, if not present put "" as default value
+    path = get_path_from_args(request)
+  
+    generic_methods = ncf.get_all_generic_methods_metadata(alt_path = path)
+    resp = jsonify( generic_methods )
+    return resp
+  except :
+    errstr = traceback.format_exc()
+    sys.stderr.write(errstr)
+    sys.stderr.flush()
+    error = jsonify ({ "error": "Unknown internal error", "details": errstr})
+    error.status_code = 500
+    return  error
 
 @app.route('/api/techniques', methods = ['POST', "PUT"])
 @check_auth
 def create_technique():
-  # Get data in JSON
-  # technique is a mandatory parameter, abort if not present
-  if not "technique" in request.json:
-    return jsonify( { 'error' : "No Technique metadata provided in the request body." } ), 400
-  else:
-    technique = request.json['technique']
-
-  if "path" in request.json and request.json['path'] != "":
-    path = request.json['path']
-  else:
-    path = default_path
-
+  """ Get data in JSON """
   try:
-    ncf.write_technique(technique,path)
-  except ncf.NcfError, ex:
-    return jsonify( { 'error' : ex.message } ), 500
-    
-  return jsonify( technique ), 201
-
+    # technique is a mandatory parameter, abort if not present
+    if not "technique" in request.json:
+      return jsonify( { 'error' : "No Technique metadata provided in the request body." } ), 400
+    else:
+      technique = request.json['technique']
+  
+    if "path" in request.json and request.json['path'] != "":
+      path = request.json['path']
+    else:
+      path = default_path
+  
+    try:
+      ncf.write_technique(technique,path)
+    except ncf.NcfError, ex:
+      return jsonify( { 'error' : ex.message } ), 500
+      
+    return jsonify( technique ), 201
+  except :
+    errstr = traceback.format_exc()
+    sys.stderr.write(errstr)
+    sys.stderr.flush()
+    error = jsonify ({ "error": "Unknown internal error", "details": errstr})
```

Comment:
Mayhe the error could be "Unknown internal error when writing technique '+technique.name (if technique name is not None!)"